packages/grid_client/src/clients/tf-grid/contracts.ts

```diff
@@ -301,13 +326,216 @@ class TFContracts extends Contracts {
+      acc.push(this.client.contracts.getContractBillingInformationByID(contract.contract_id));
+      return acc;
+    }, []);
+    const overDraftPromises = contracts.reduce((acc: Promise<ContractPaymentState>[], contract) => {
+      acc.push(this.client.contracts.getContractPaymentState(contract.contract_id));
+      return acc;
+    }, []);
+
+    const overDraftResult = await Promise.all(overDraftPromises);
+    const billingInfoResult = await Promise.all(BillingInformationPromises);
+
+    const totalOverDraft = overDraftResult.reduce(
+      (acc: Decimal, paymentState) => acc.add(paymentState.additionalOverdraft).add(paymentState.standardOverdraft),
+      new Decimal(0),
+    );
+    const totalNUCost = billingInfoResult.reduce((acc: number, billingInfo) => acc + billingInfo.amountUnbilled, 0);
+    const totalIPCost = (ipPrice * (contracts.length || 0)) / 1e7;
+    return {
+      //return ip cost per month
+      totalIpCost: totalIPCost * 24 * 30,
+      totalOverdraft: totalOverDraft.add(totalNUCost),
+    };
+  }
+
+  /**
+   * This function is for calculating the estimated cost of the contract per month.
+   * @description
+   *  Name contract cost is fixed price for the unique name,
+   * Rent contract cost is the cost of the total node resources,
+   * Node contract have to cases:
+   * 1- on rented contract, this case the cost will be only for the ipv4.
+   * 2- on shared node, this will be the shared price of the used resources
+
+   * @param {Contract} contract 
+   * @param {GridProxyClient} proxy 
+   * @returns the cost of the contract per month in USD
+   */
+  async getContractCost(contract: Contract, proxy: GridProxyClient) {
+    const calc = new calculator(this.client);
+
+    if (contract.type == ContractType.Name) return await calc.namePricing();
+
+    //TODO allow ipv4 to be number
+
+    /** Other contract types need the node information */
+    const nodeDetails = await proxy.nodes.byId(contract.details.nodeId);
+
+    const certified = nodeDetails.certificationType == CertificationType.Certified ? true : false;
+    if (contract.type == ContractType.Rent) {
+      const { cru, sru, mru, hru } = nodeDetails.total_resources;
+      const USDCost = (
+        await calc.calculateWithMyBalance({
+          ipv4u: false,
+          certified,
+          cru: bytesToGB(cru),
+          mru: bytesToGB(mru),
+          hru: bytesToGB(hru),
+          sru: bytesToGB(sru),
+        })
+      ).dedicatedPrice;
+
+      return USDCost;
+    }
+
+    /** Node Contract */
+
+    /** Node contract on rented node
+     * If the node contract has IPV4 will return the price of the ipv4 per month
+     * If not there is no cost, will return zero
+     */
+
+    if (nodeDetails.rented) {
+      if (!contract.details.number_of_public_ips) return 0;
+
+      const ipPrice = (await this.client.pricingPolicies.get({ id: 1 })).ipu.value / 10 ** 7;
```

Comment:
I would avoid using magic numbers like 10 ** 7 and define it as constant `TFT_CONVERSION_FACTOR` (maybe in the client) then use it everywhere else for clarity and maintainability.